ad7768-1fmcz/ad77681.c

```diff
@@ -282,13 +665,291 @@ int32_t ad77681_set_power_mode(struct ad77681_dev *dev,
+		AD77681_ANALOG_REF_BUF_POS(REFp));
+
+	dev->REFp = REFp;
+
+	return ret;
+}
+
+/**
+ * Set filter type and decimation ratio
+ * @param dev 			The device structure.
+ * @param decimate		Decimation ratio of filter
+ *						Accepted values: AD77681_SINC5_FIR_DECx32
+ *										 AD77681_SINC5_FIR_DECx64
+ *										 AD77681_SINC5_FIR_DECx128
+ *										 AD77681_SINC5_FIR_DECx256
+ *										 AD77681_SINC5_FIR_DECx512
+ *										 AD77681_SINC5_FIR_DECx1024
+ * @param filter    	Select filter type
+ *						Accepted values: AD77681_SINC5
+ *										 AD77681_SINC5_DECx8
+ *										 AD77681_SINC5_DECx16
+ *										 AD77681_SINC3
+ *										 AD77681_FIR
+ * @param sinc3_osr		Select decimation ratio for SINC3 filter separately as integer
+ *						from 0 to 8192.
+ *						See the AD7768-1 datasheet for more info
+ * @return 0 in case of success, negative error code otherwise.
+ */
+
+int32_t ad77681_set_filter_type(struct ad77681_dev *dev,
+	enum ad77681_sinc5_fir_decimate decimate,
+	enum ad77681_filter_type filter,
+	uint16_t sinc3_osr)
+{
+	int32_t ret = 0;
+	
+	ret |= ad77681_spi_reg_write(dev, AD77681_REG_DIGITAL_FILTER, 0x00);
+	
+	if ((filter == AD77681_SINC5_DECx8) || (filter == AD77681_SINC5_DECx16)) {			// SINC5 for OSR 8x and 16x
+		ret |= ad77681_spi_write_mask(dev,
+			AD77681_REG_DIGITAL_FILTER,
+			AD77681_DIGI_FILTER_FILTER_MSK,
+			AD77681_DIGI_FILTER_FILTER(filter));
+	}
+	else if ((filter == AD77681_SINC5) || (filter == AD77681_FIR)) {					// SINC5 and FIR for osr 32x to 1024x
+		ret |= ad77681_spi_write_mask(dev,
+			AD77681_REG_DIGITAL_FILTER,
+			AD77681_DIGI_FILTER_FILTER_MSK,
+			AD77681_DIGI_FILTER_FILTER(filter));
+		
+		ret |= ad77681_spi_write_mask(dev,
+			AD77681_REG_DIGITAL_FILTER,
+			AD77681_DIGI_FILTER_DEC_RATE_MSK,
+			AD77681_DIGI_FILTER_DEC_RATE(decimate));
+	}
+
+	else {																				// SINC3
+		uint8_t sinc3_LSB = 0, sinc3_MSB = 0;
+		
+		sinc3_MSB = sinc3_osr >> 8;
+		sinc3_LSB = sinc3_osr & 0x00FF;
+		
+		ret |= ad77681_spi_write_mask(dev,
+			AD77681_REG_DIGITAL_FILTER,
+			AD77681_DIGI_FILTER_FILTER_MSK,
+			AD77681_DIGI_FILTER_FILTER(filter));
+		
+		ret |= ad77681_spi_write_mask(dev,
+			AD77681_REG_SINC3_DEC_RATE_MSB,
+			AD77681_SINC3_DEC_RATE_MSB_MSK,
+			AD77681_SINC3_DEC_RATE_MSB(sinc3_MSB));
+		
+		ret |= ad77681_spi_write_mask(dev,
+			AD77681_REG_SINC3_DEC_RATE_LSB,
+			AD77681_SINC3_DEC_RATE_LSB_MSK,
+			AD77681_SINC3_DEC_RATE_LSB(sinc3_LSB));	
+	}
+	
+	dev->decimate = decimate;
+	dev->filter = filter;
+	ret |= ad77681_initiate_sync(dev);															// Sync pulse after each filter change
+	return ret;
+}
+
+/**
+ * Enable 50/60 Hz rejection
+ * @param dev   -  The device structure.
+ * @param enable - The positive reference buffer selector
+ * 					Accepted values: true
+ *									 false
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad77681_set_50HZ_rejection(struct ad77681_dev *dev,
+	uint8_t enable)
+{
+	int32_t ret;
```

Comment:
maybe directly `return ad77681_spi_write_mask`.

No need for extra variable `ret` in this situation.